common/src/main/java/org/apache/celeborn/common/network/client/TransportClient.java

```diff
@@ -415,29 +437,29 @@ protected void handleFailure(String errorMsg, Throwable cause) {
 
   private class PushChannelListener extends StdChannelListener {
     final long pushRequestId;
-    Runnable rpcSendOutCallback;
+    Runnable rpcFailureCallback;
 
     PushChannelListener(long pushRequestId) {
       this(pushRequestId, null);
     }
 
-    PushChannelListener(long pushRequestId, Runnable rpcSendOutCallback) {
+    PushChannelListener(long pushRequestId, Runnable rpcFailureCallback) {
       super("PUSH " + pushRequestId);
       this.pushRequestId = pushRequestId;
-      this.rpcSendOutCallback = rpcSendOutCallback;
+      this.rpcFailureCallback = rpcFailureCallback;
     }
 
     @Override
     public void operationComplete(Future<? super Void> future) throws Exception {
       super.operationComplete(future);
-      if (rpcSendOutCallback != null) {
```

Comment:
This changes behavior, is it OK? cc @RexXiong @FMX

Reply:
IIRC this will affect buffer release from flink, IMO we can retain the current setup and remove the rpcFailureCallback call from handleFailure, since operationComplete will be invoked regardless of whether the channel fails

Reply:
How about current?

Reply:
> How about current?

Make sense.